File: solutions-in-javascript/staircase.js
function staircase(n) {
    for (let i = 1; i <= n; i++) {
      // Calculate spaces for right alignment
      const spaces = ' '.repeat(n - i);
      // Calculate hashes for this row
      const hashes = '#'.repeat(i);
      // Print the row with right alignment
      console.log(spaces + hashes);
    }
  }
  
  // Test with n = 6
  staircase(6);
  

File: solutions-in-python/staircase.py
def staircase(n):
    for i in range(1, n + 1):
        # Calculate the number of spaces needed for right alignment
        spaces = ' ' * (n - i)
        # Calculate the number of hash symbols for this row
        hashes = '#' * i
        # Print the row with right alignment
        print(spaces + hashes)